index.html

```diff
@@ -63,17 +66,60 @@ <h1>liczba wyborcow w danych okregach wyborczych</h1>
                     var selectedDistrict = rows[selectedOption].split(';')[0];
 
                     var districtRows = rows.filter(function (row) {
-                    return row.startsWith(selectedDistrict + ';');
+                        return row.startsWith(selectedDistrict + ';');
                     });
 
                     var squaredDistance = 0;
                     for (var i = 0; i < districtRows.length; i++) {
-                    var rowData = districtRows[i].split(';');
-                    var votersCount = parseFloat(rowData[4].replace(',', '.')) || 0;
-                    squaredDistance += Math.pow(votersCount, 2);
+                        var rowData = districtRows[i].split(';');
+                        var votersCount = parseFloat(rowData[4].replace(',', '.')) || 0;
+                        squaredDistance += Math.pow(votersCount, 2);
                     }
 
                     document.getElementById('kwadratodleglosci').innerHTML = 'Kwadrat odległości dla Okręgu Wyborczego ' + selectedDistrict + ': ' + squaredDistance;
+
+                    findClosestDistricts(selectedDistrict, squaredDistance);
+                };
+
+                reader.readAsText(loadedFile);
+            } else {
+                alert('Proszę najpierw załadować plik CSV i wybrać okręg.');
+            }
+        }
+
+        function findClosestDistricts(selectedDistrict, currentSquaredDistance) {
+            var selectedOption = document.getElementById('select').value;
+            if (loadedFile && selectedOption !== "") {
+                var reader = new FileReader();
+                reader.onload = function (e) {
+                    var csvData = e.target.result;
+                    var rows = csvData.split('\n');
+
+                    var distances = [];
+
+                    for (var i = 0; i < addedDistricts.length; i++) {
+                        var districtNumber = addedDistricts[i];
+                        if (districtNumber !== selectedDistrict) {
+                            var districtRows = rows.filter(function (row) {
+                                return row.startsWith(districtNumber + ';');
+                            });
+
+                            var squaredDistance = 0;
+                            for (var j = 0; j < districtRows.length; j++) {
+                                var rowData = districtRows[j].split(';');
+                                var votersCount = parseFloat(rowData[4].replace(',', '.')) || 0;
+                                squaredDistance += Math.pow(votersCount, 2);

```

Comment:
The number of voters is different in each district. For the final solution, we have to use percentages instead of the number of voters, because we want to compare the tendencies between districts, not just absolute numbers of voters.